### PR TITLE
feat: add macro for scoped probes

### PR DESCRIPTION
**Pull request by markaylett**
Add a macro that provides a convenient RAII-style mechanism for begin and end probes.

DEV-3489

### CODE REVIEW

File: toolbox/sys/Trace.hpp [hunk 1]
```diff
@@ -22,8 +22,19 @@
...
 #define TOOLBOX_PROBE(provider, name, ...) STAP_PROBEV(provider, name, ##__VA_ARGS__)
+// clang-format off
+#define TOOLBOX_PROBE_SCOPED(provider, name)                \
+    TOOLBOX_PROBE(provider, name ## _begin);                \
+    const struct __Finally_ ## provider ## _ ## name {      \
+        ~__Finally_ ## provider ## _ ## name () noexcept    \
+        {                                                   \
+            TOOLBOX_PROBE(provider, name ## _end);          \
+        }                                                   \
+    } __finally_ ## provider ## _ ## name
```
Comment (Shivix) [inline toolbox/sys/Trace.hpp:33]: missing semicolon after name

Comment (markaylett) [reply toolbox/sys/Trace.hpp:33]: This is intentional. It ensures the consistent use of semi-colons are macro calls:
```c++
TOOLBOX_PROBE_SCOPED(reactor, dispatch); // <= Semi-colon required.
```

Comment (Shivix) [reply toolbox/sys/Trace.hpp:33]: ah, I did consider this, wasn't sure though.

**Review by Shivix — APPROVED** [review]

File: toolbox/io/Reactor.cpp [hunk 2]
```diff
@@ -89,15 +89,14 @@ int Reactor::poll(CyclTime now, Duration timeout)
...
-    TOOLBOX_PROBE(reactor, enter_cycle);
+    TOOLBOX_PROBE_SCOPED(reactor, dispatch);
     auto work{tqs_[High].dispatch(now)};
     work += dispatch(now, buf, n);
     // Low priority timers are only dispatched during empty cycles.
     if (work == 0) {
         work += tqs_[Low].dispatch(now);
     }
     io::dispatch(now, hooks_);
-    TOOLBOX_PROBE(reactor, exit_cycle, work);
```
Comment (markaylett) [inline toolbox/io/Reactor.cpp:100]: Bug: the new version omits the "work" argument. I will push a fix shortly.

**Pull request merged**